src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -160,101 +165,111 @@ const UsersFragment = ({
       }
+    });
 
-      const followingUsers = [];
-      const unfollowingUsers = [];
-
-      users.forEach((item) => {
-        if (item.user?.user_id_follower || item.user_id_follower) {
-          followingUsers.push(item);
-        } else {
-          unfollowingUsers.push(item);
-        }
-      });
-
-      return [
-        followingUsers.map((item, index) =>
-          renderDiscoveryItem(FROM_FOLLOWED_USERS_INITIAL, 'followedUsers', item, index)
-        )
-      ]
-        .concat([
-          route.name !== 'Followings' && (
-            <DiscoveryTitleSeparator key="user-title-separator" text="Suggested Users" />
-          )
-        ])
-        .concat([
-          unfollowingUsers.map((item, index) =>
-            renderDiscoveryItem(
-              FROM_FOLLOWED_USERS_INITIAL,
-              'followedUsers',
-              item,
-              index + followingUsers.length
-            )
-          )
-        ]);
-    }
+    const data = isFirstTimeOpen
+      ? withoutRecent
+        ? initialUsers.length !== 0
+          ? initialUsers
+          : users
+        : [...initFollowingUsers, {separator: true}, ...initUnfollowingUsers]
+      : unfollowedUsers.length !== 0
+      ? [...followedUsers, {separator: true}, ...unfollowedUsers]
+      : followedUsers;
 
     return (
-      <>
-        {followedUsers.map((item, index) =>
-          renderDiscoveryItem(FROM_FOLLOWED_USERS, 'followedUsers', item, index)
-        )}
-
-        {route.name !== 'Followings' &&
-          unfollowedUsers.length > 0 &&
-          followedUsers.length > 0 &&
-          !withoutRecent && (
-            <View style={styles.unfollowedHeaderContainer}>
-              <Text style={styles.unfollowedHeaders}>{StringConstant.discoveryMoreUsers}</Text>
-            </View>
-          )}
-        {route.name !== 'Followings' &&
-          unfollowedUsers.map((item, index) =>
-            renderDiscoveryItem(FROM_UNFOLLOWED_USERS, 'unfollowedUsers', item, index)
-          )}
-      </>
+      <FlatList
+        onMomentumScrollBegin={handleScroll}
+        contentContainerStyle={{paddingBottom: 100}}
+        data={data || []}
+        renderItem={({index, item}) => {
+          return renderDiscoveryItem({
+            from: isFirstTimeOpen
+              ? withoutRecent
+                ? initialUsers.length !== 0
+                  ? FROM_USERS_INITIAL
+                  : FROM_FOLLOWED_USERS_INITIAL
+                : FROM_FOLLOWED_USERS_INITIAL
+              : unfollowedUsers.length !== 0
+              ? index > followedUsers.length
+                ? FROM_UNFOLLOWED_USERS
+                : FROM_FOLLOWED_USERS
+              : index > followedUsers.length
+              ? FROM_UNFOLLOWED_USERS
+              : FROM_FOLLOWED_USERS,
+            item,
+            index
+          });
+        }}
+        keyExtractor={(item, index) => index.toString()}
+        onEndReached={() => fetchData()}
+        onEndReachedThreshold={0.6}
+      />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for rendering items has been significantly refactored. The `renderRecentSearch`, `renderDiscoveryItem`, and `renderUsersItem` functions have been updated to accommodate the new props and state variables. The use of `FlatList` for rendering the list of users is a good practice for performance optimization. However, ensure that the data manipulation logic in `renderUsersItem` is correct and efficient.
```diff
- const renderDiscoveryItem = (from, key, item, index) => {
...
- const renderUsersItem = () => {
...
+ const renderRecentSearch = (index) => {
...
+ const renderDiscoveryItem = ({from, item, index}) => {
...
+ const renderUsersItem = () => {
...
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->